crates/rustc_codegen_spirv/src/abi.rs

```diff
@@ -582,6 +585,20 @@ fn get_storage_class<'tcx>(cx: &CodegenCx<'tcx>, ty: TyAndLayout<'tcx>) -> Optio
     None
 }
 
+/// Handles `#[spirv(block)]`. Note this is only called in the scalar translation code, because this is only
+/// used for spooky builtin stuff, and we pinky promise to never have more than one pointer field in one of these.
+// TODO: Enforce this is only used in spirv-std.
```

Comment:
This comment isn't correct, right? It's the opposite of what's actually happening, it's only called in the _ADT_ translation code. I believe this should be called at the entrypoint of `trans_type_impl`, in the same for loop that `trans_image` is called, and error when applied to a struct that is a scalar/scalar pair/etc.